From d2c95b0e62f91e3d47e8fd7e1300a00d332bd9a2 Mon Sep 17 00:00:00 2001
From: rocky <rocky@gnu.org>
Date: Sun, 20 Dec 2009 12:53:32 -0500
Subject: [PATCH 4/6] Looks like sig line fixes back ported to bash 4.0. No need for a SunOS-specific file.

test-misc and test-setshow are derived from .in files.
---
 test/data/sig-SunOS.right     |   63 -----------------------------------------
 test/integration/test-misc    |   35 ----------------------
 test/integration/test-setshow |   26 -----------------
 test/integration/test-sig     |    4 +--
 4 files changed, 1 insertions(+), 127 deletions(-)
 delete mode 100644 test/data/sig-SunOS.right
 delete mode 100755 test/integration/test-misc
 delete mode 100755 test/integration/test-setshow

diff --git a/test/data/sig-SunOS.right b/test/data/sig-SunOS.right
deleted file mode 100644
index 4656d84..0000000
--- a/test/data/sig-SunOS.right
+++ /dev/null
@@ -1,63 +0,0 @@
-(sig.sh:7):
-7:	if [[ "$1"x != x ]] ; then 
-+# Test that debugged program's signals are saved and restored across
-+# debug calls.
-+### Testing that we have our signal set up...
-+info signals
-Signal       Stop   Print   Stack     Value
-EXIT         stop   noprint nostack   trap -- '_Dbg_exit_handler "$BASH_COMMAND"' EXIT
-EXIT         stop   noprint nostack   trap -- '_Dbg_exit_handler "$BASH_COMMAND"' EXIT
-SIGINT       stop   print   showstack trap -- '_Dbg_sig_handler 2 "$BASH_COMMAND" "$@"' SIGINT
-SIGQUIT      stop   print   showstack trap -- '_Dbg_sig_handler 3 "$BASH_COMMAND" "$@"' SIGQUIT
-SIGILL       stop   print   showstack trap -- '_Dbg_sig_handler 4 "$BASH_COMMAND" "$@"' SIGILL
-SIGTRAP      stop   print   showstack trap -- '_Dbg_sig_handler 5 "$BASH_COMMAND" "$@"' SIGTRAP
-SIGTERM      stop   print   showstack trap -- '_Dbg_sig_handler 15 "$BASH_COMMAND" "$@"' SIGTERM
-+### Testing handle command...
-+handle TERM nostack
-+handle foo
-Bad signal name: foo
-+handle 1000
-Bad signal number: 1000
-+handle TERM bogus
-Need to give a command: stop, nostop, stack, nostack, print, noprint
-+eval kill -TERM $$
-Program received signal SIGTERM (15)...
-+### Should not have printed a stack trace above...
-+handle TERM noprint
-+handle TERM stack
-+handle INT nostop
-+eval kill -TERM $$
-+info signals
-Signal       Stop   Print   Stack     Value
-EXIT         stop   noprint nostack   trap -- '_Dbg_exit_handler "$BASH_COMMAND"' EXIT
-EXIT         stop   noprint nostack   trap -- '_Dbg_exit_handler "$BASH_COMMAND"' EXIT
-SIGINT       nostop print   showstack trap -- '_Dbg_sig_handler 2 "$BASH_COMMAND" "$@"' SIGINT
-SIGQUIT      stop   print   showstack trap -- '_Dbg_sig_handler 3 "$BASH_COMMAND" "$@"' SIGQUIT
-SIGILL       stop   print   showstack trap -- '_Dbg_sig_handler 4 "$BASH_COMMAND" "$@"' SIGILL
-SIGTRAP      stop   print   showstack trap -- '_Dbg_sig_handler 5 "$BASH_COMMAND" "$@"' SIGTRAP
-SIGTERM      stop   noprint showstack trap -- '_Dbg_sig_handler 15 "$BASH_COMMAND" "$@"' SIGTERM
-+continue 
-Program received signal SIGTERM (15)...
-->0 in file `eval.sh' at line 1
-##1 _Dbg_do_eval("kill", "-TERM", "$$") called from file `processor.sh' at line 402
-##2 _Dbg_onecmd("eval", "kill -TERM $$") called from file `processor.sh' at line 167
-##3 _Dbg_process_commands() called from file `hook.sh' at line 188
-##4 _Dbg_debug_trap_handler("0", "[[ "$1"x != x ]]") called from file `sig.sh' at line 7
-##5 source("sig.sh") called from file `bashdb' at line 157
-##6 main() called from file `bashdb' at line 0
-+### Should have printed a stack trace above...
-+continue 
-+where 1
-->0 in file `sig.sh' at line 1
-+continue 
-Program received signal SIGINT (2)...
-->0 in file `sig.sh' at line 1
-##1 source("sig.sh") called from file `bashdb' at line 157
-##2 main() called from file `bashdb' at line 0
-Program received signal SIGINT (2)...
-->0 in file `sig.sh' at line 1
-##1 source("sig.sh") called from file `bashdb' at line 157
-##2 main() called from file `bashdb' at line 0
-Debugged program terminated normally. Use q to quit or R to restart.
-+kill 
-sig.tests: line 12: 28364 Killed                  $SH ${top_builddir}/bashdb -B -q -L $top_builddir -x $cmdfile $debugged_script
diff --git a/test/integration/test-misc b/test/integration/test-misc
deleted file mode 100755
index 80f387a..0000000
--- a/test/integration/test-misc
+++ /dev/null
@@ -1,35 +0,0 @@
-#!/opt/csw/bin/bash
-# -*- shell-script -*-
-#$Id: test-misc,v 1.3 2008/08/23 17:09:36 rockyb Exp $
-
-[ -z "$srcdir" ] && export srcdir=$PWD
-[ -z "$builddir" ] && export builddir=$PWD
-[ -z "$top_builddir" ] && export top_builddir=$builddir/../..
-
-TEST_NAME='misc'
-
-. ${top_builddir}/test/integration/check-common.sh
-
-debugged_script="$top_builddir/test/example/dbg-test2.sh"
-
-(cd $srcdir && run_debugger $debugged_script 2>&1 >$TEST_FILE </dev/null)
-
-# Different OS's handle ARGV differently, and "info args" may come out
-# differently depending on that.
-cat ${TEST_FILE} | /opt/csw/bin/gsed -e "s:1 = .*/dbg-test2.sh:1 = ./example/dbg-test2.sh:" \
-| /opt/csw/bin/gsed -e 's:record the command history is .*:record the command history is: ' \
-> ${TEST_FILTERED_FILE}
-
-if (( BASH_VERSINFO[0] > 4 || \
-    (BASH_VERSINFO[0] == 4 && BASH_VERSINFO[1] >= 1) )) ; then 
-    RIGHT_FILE="${top_srcdir}/test/data/${TEST_NAME}-output-41.right"
-fi
-
-check_output $TEST_FILTERED_FILE $RIGHT_FILE
-rc=$?
-
-((0 != rc)) && exit $rc
-rm -f $TEST_FILE
-
-# Return code tells testing mechanism whether passed or not.
-exit 0
diff --git a/test/integration/test-setshow b/test/integration/test-setshow
deleted file mode 100755
index deeb3ba..0000000
--- a/test/integration/test-setshow
+++ /dev/null
@@ -1,26 +0,0 @@
-#!/opt/csw/bin/bash -f
-# -*- shell-script -*-
-#$Id: test-setshow,v 1.3 2008/08/23 17:09:36 rockyb Exp $
-
-[ -z "$srcdir" ] && export srcdir=$PWD
-[ -z "$builddir" ] && export builddir=$PWD
-[ -z "$top_builddir" ] && export top_builddir=$builddir/../..
-
-TEST_NAME='setshow'
-
-. ${top_builddir}/test/integration/check-common.sh
-
-(cd $srcdir && $SH ${srcdir}/${TEST_NAME}.tests > $TEST_FILE 2>&1 < /dev/null)
-
-cat ${TEST_FILE} | /opt/csw/bin/gsed -e "s:-L .* -x .*/data/setshow\.cmd .*/example/dbg-test2.sh:-L .. -x data/setshow.cmd example/dbg-test2.sh:" \
-| /opt/csw/bin/gsed -e 's:record the command history is .*:record the command history is: ' \
-> ${TEST_FILTERED_FILE}
-
-check_output $TEST_FILTERED_FILE $RIGHT_FILE
-rc=$?
-
-((0 != rc)) && exit $rc
-rm -f $TEST_FILE
-
-# Return code tells testing mechanism whether passed or not.
-exit 0
diff --git a/test/integration/test-sig b/test/integration/test-sig
index 98b2359..1a69609 100755
--- a/test/integration/test-sig
+++ b/test/integration/test-sig
@@ -11,10 +11,8 @@ TEST_NAME='sig'
 . ${top_builddir}/test/integration/check-common.sh
 
 if (( BASH_VERSINFO[0] > 4 || \
-    (BASH_VERSINFO[0] == 4 && BASH_VERSINFO[1] >= 1) )) ; then 
+    (BASH_VERSINFO[0] == 4 && BASH_VERSINFO[1] >= 0) )) ; then 
     RIGHT_FILE="${top_srcdir}/test/data/${TEST_NAME}-41.right"
-elif [[ $(uname -s) == 'Darwin' ]] ; then
-    RIGHT_FILE="${top_srcdir}/test/data/${TEST_NAME}-Darwin.right"
 elif [[ $(uname -s) == 'SunOS' ]] ; then
     RIGHT_FILE="${top_srcdir}/test/data/${TEST_NAME}-SunOS.right"
 fi
-- 
1.6.2.1

